ASFAchievementManager/AchievementHandler.cs

```diff
@@ -21,53 +19,52 @@ public sealed class AchievementHandler : ClientMsgHandler {
 
 			switch (packetMsg.MsgType) {
 				case EMsg.ClientGetUserStatsResponse:
-					HandleGetUserStatsResponse(packetMsg);
-
+					ClientMsgProtobuf<CMsgClientGetUserStatsResponse> getAchievementsResponse = new(packetMsg);
+					Client.PostCallback(new GetAchievementsCallback(packetMsg.TargetJobID, getAchievementsResponse.Body));
 					break;
 				case EMsg.ClientStoreUserStatsResponse:
-					HandleStoreUserStatsResponse(packetMsg);
-
+					ClientMsgProtobuf<CMsgClientStoreUserStatsResponse> setAchievementsResponse = new(packetMsg);
+					Client.PostCallback(new SetAchievementsCallback(packetMsg.TargetJobID, setAchievementsResponse.Body));
 					break;
 			}
 
 		}
 
-		private void HandleGetUserStatsResponse(IPacketMsg packetMsg) {
-			if (packetMsg == null) {
-				ASF.ArchiLogger.LogNullError(nameof(packetMsg));
-				return;
-			}
-			ClientMsgProtobuf<CMsgClientGetUserStatsResponse> response = new ClientMsgProtobuf<CMsgClientGetUserStatsResponse>(packetMsg);
-			if (!Responses.TryAdd(response.Body.game_id, new StoredResponse {
-				Success = response.Body.eresult == 1,
-				Response = response.Body
-			})) {
-				ASF.ArchiLogger.LogGenericError(string.Format(Strings.ErrorFailingRequest, "GetAchievements"));
-			}
+		internal abstract class AchievementsCallBack<T> : CallbackMsg {
+			internal readonly T Response;
+			internal readonly bool Success;
 
-			if (response.Body.eresult != 1) {
-				ASF.ArchiLogger.LogGenericError(string.Format(Strings.ErrorFailingRequest, "GetAchievements"));
-			}
+			internal AchievementsCallBack(JobID jobID, T msg, Func<T, EResult> eresultGetter, string error) {
```

Comment:
I wouldn't consider an error message to be something for the constructor. Why not an abstract property or `GetType().Name`?

Reply:
@Abrynos, could you please elaborate what exactly you proposed here (implying that you still remember this, lol)?